### PR TITLE
Enhancement #8005: Open rocketcdn modal query string

### LINKED ISSUE #8005: Open rocket-cdn iframe using a URL with a query string or '#' fragment
**Context:**
Currently RocketCDN tab url is `wp-admin/options-general.php?page=wprocket#page_cdn`

With this URL, a redirect to customer dashboard can't open RocketCDN.

We want to redirect the client if he uses PayPal or if we face a problem in rocketcdn website configuration during express checkout to the roctetcdn-iframe. In this scenario, we want to **Reconfigure** RocketCDN.

We don't deprecate RocketCDN iframe to handle cases of customer that can't purchase RocketCDN with One Click Purchase.
That's why, a redirect (from WPR website) needs to be able to open the iframe.

**Expected behavior:**
`wp-admin/options-general.php?page=wprocket&rocketcdn_open_iframe=1` or
`wp-admin/options-general.php?page=wprocket#page_cdn`
should open the iframe after a redirect

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #8005 

This change allows the RocketCDN iframe modal to be automatically opened via a URL parameter. When users are redirected from the WP Rocket website (e.g., after PayPal checkout or when configuration issues occur during express checkout), the modal will open automatically to allow RocketCDN reconfiguration.

Users visiting `wp-admin/options-general.php?page=wprocket&rocketcdn_open_iframe=1` will now see the RocketCDN modal open automatically, streamlining the checkout and configuration flow.

## Type of change

- [x] New feature (non-breaking change which adds functionality).

### What was tested

**Manual Testing:**
1. ✅ Accessing WP Rocket settings with `rocketcdn_open_iframe=1` parameter opens the modal automatically
2. ✅ URL is cleaned after modal opens (parameter removed via `history.replaceState`)
3. ✅ Page refresh after modal opens does not re-trigger the modal
4. ✅ Modal opens correctly and iframe loads as expected
5. ✅ Normal modal opening functionality (without parameter) still works
6. ✅ Other URL parameters are preserved after cleanup

### How to test

**Prerequisites:**
- WordPress site with WP Rocket installed
- Access to wp-admin

**Steps to test:**
1. Navigate to: `wp-admin/options-general.php?page=wprocket&rocketcdn_open_iframe=1`
2. **Expected:** RocketCDN modal should open automatically
3. **Expected:** URL bar should update to remove the `rocketcdn_open_iframe` parameter (e.g., `wp-admin/options-general.php?page=wprocket#page_cdn`)
4. Refresh the page
5. **Expected:** Modal should NOT re-open automatically
6. Test with additional parameters: `wp-admin/options-general.php?page=wprocket&rocketcdn_open_iframe=1&test=value#page_cdn`
7. **Expected:** Modal opens, URL becomes `wp-admin/options-general.php?page=wprocket&test=value#page_cdn` (only `rocketcdn_open_iframe` removed)
8. Test normal modal opening via UI elements still works

### Affected Features & Quality Assurance Scope

**Affected Features:**
- RocketCDN modal opening functionality
- URL parameter handling in WP Rocket settings page
- RocketCDN checkout flow (integration with WP Rocket website redirects)

**QA Testing Scope:**
- All existing RocketCDN modal interactions (open, close, form submissions)
- URL parameter handling and browser history navigation
- Checkout flow redirects from wp-rocket.me
- PayPal checkout flow edge cases
- Express checkout error handling flows

### Documentation

**Implementation:**

Added a new function `maybeOpenModalFromURL()` in `src/js/global/rocketcdn.js` that:

1. **Detects URL parameter:** Uses `URLSearchParams` to check for `rocketcdn_open_iframe=1` query parameter
2. **Opens modal:** Calls `MicroModal.show('wpr-rocketcdn-modal')` when parameter is present
3. **Cleans URL:** Uses `window.history.replaceState()` to remove the parameter from the URL without triggering a page reload

**Code flow:**
```
Page Load (DOMContentLoaded)
    ↓
maybeOpenModalFromURL() executes
    ↓
Check for rocketcdn_open_iframe=1 parameter
    ↓
If present:
    ├─→ Open modal
    ├─→ Remove parameter from URL
    └─→ Update browser history
```

**Pattern used:**
Follows the same architectural pattern as Rocket Insights URL parameter handling (see `src/js/global/ajax.js` for similar implementation)

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

## Unticked items justification

**Built-in tests:** The change is a small client-side JavaScript enhancement for URL parameter detection. 

# Additional Checks
- [x] In the case of complex code, I wrote comments to explain it.
- [ ] When possible, I prepared ways to observe the implemented system (logs, data, etc.).
- [x] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

**Additional checks justification:**
- Comments added to explain URL cleanup purpose
- No logging needed for this simple client-side feature (would add noise without value)
- No error-prone operations (standard browser APIs used, already validated by browser vendors)

### CODE REVIEW

File: src/js/global/rocketcdn.js [hunk 1]
```diff
@@ -98,6 +99,19 @@
...
 
+	function maybeOpenModalFromURL() {
+		const urlParams = new URLSearchParams( window.location.search );
+
+		if ( urlParams.has( 'rocketcdn_open_iframe' ) && '1' === urlParams.get( 'rocketcdn_open_iframe' ) ) {
+			MicroModal.show( 'wpr-rocketcdn-modal' );
+
+			// Clean up the URL to prevent re-triggering on refresh
+			urlParams.delete( 'rocketcdn_open_iframe' );
+			const newURL = window.location.pathname + '?' + urlParams.toString() + window.location.hash;
```
[the remark this answers is not in the document]
Comment (wordpressfan) [reply src/js/global/rocketcdn.js]: I know this is not our case because we know that this page has query string `?page=wprocket` but it's a good practice to apply this small enhancement.

**Review by wordpressfan — APPROVED** [review]
Only one comment, this means that the modal will open but the user is still on dashboard not in cdn tab, I don't know if it's expected or not but I think it'd be better to redirect him to the page_cdn hash

**Comment by Mai-Saad** [comment]
@Miraeld Thanks for the PR. 
With production saas:

- [x] buy CDN from normal banner  
- [x] Cancel CDN subscription then delete all transient and refresh admin, CDN banner is there  
- [x] buy CDN using the new URL  
- [x] open new URL while user already have CDN => related thread https://group-onecom.slack.com/archives/C08F4LZB2QG/p1770974874602289?thread_ts=1770974634.852009&cid=C08F4LZB2QG

Note: 
1- out of test scope (should be tested on saas @wp-media/engineering-wpr-addons-team )
- Checkout flow redirects from wp-rocket.me
- PayPal checkout flow edge cases
- Express checkout error handling flows

2- The URL to be used from saas https://group-onecom.slack.com/archives/C08F4LZB2QG/p1770978180421029?thread_ts=1770976390.072789&cid=C08F4LZB2QG @mostafa-hisham
